identity-iota/src/tangle/publish.rs

```diff
@@ -0,0 +1,264 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use identity_did::verification::MethodRef;
+use identity_did::verification::MethodType;
+
+use crate::did::IotaDocument;
+
+// Method types allowed to sign a DID document update.
+pub const UPDATE_METHOD_TYPES: &[MethodType] = &[MethodType::Ed25519VerificationKey2018];
+
+/// Determines whether an updated document needs to be published as an integration or diff message.
+#[derive(Clone, Copy, Debug)]
+pub enum PublishType {
+  Integration,
+  Diff,
+}
+
+impl PublishType {
+  /// Compares two versions of a document and returns whether it needs to be published
+  /// as an integration or diff message. If `None` is returned, no update is required.
+  ///
+  /// Note: A newly created document must always be published as an integration message, and
+  /// this method does not handle this case.
+  pub fn new(old_doc: &IotaDocument, new_doc: &IotaDocument) -> Option<PublishType> {
+    if old_doc == new_doc {
+      return None;
+    }
+
+    let old_capability_invocation_set = old_doc
+      .as_document()
+      .capability_invocation()
+      .iter()
+      .map(|method_ref| match method_ref {
+        MethodRef::Embed(method) => Some(method),
+        MethodRef::Refer(did_url) => old_doc.as_document().resolve_method(did_url),
+      })
+      .filter(|method| {
+        if let Some(method) = method {
+          UPDATE_METHOD_TYPES.contains(&method.key_type())
+        } else {
+          true
+        }
+      })
```

Comment:
Oh, I thought this is what you meant in [this comment](https://github.com/iotaledger/identity.rs/pull/462#discussion_r747741254). I think if we decide that `MerkleKeyCollection`s will never be able to sign document updates, then it's fine. Any new signature type we introduce, we have to make a decision for whether to add or leave it out of `UPDATE_METHOD_TYPES`.

Reply:
It was ^^ I was just questioning my own decision since the ramifications for future breaking changes didn't occur to me at the time.

Absolutely fine leaving it as-is for now to be clear.